MAST: remove deprecated code, fix tests, general cleanup

ceb8 opened the pull request:
This PR includes a number of cleanup tasks, primarily in the MAST codebase. With the exception of the documentation updates, most of the changes should be transparent to the user.

Changes included in this PR:

- Removal of deprecated MAST authentication code: Since MAST's transition to the new authentication system is complete I have removed all the code that was for the old system.
   - Remove all shibboleth login functionality
   - Remove password method instructions 
   - Deprecated `get_token` method since tokens are now the "secret" for authentication
   - Updated doc strings as needed
- Removed Tesscut livecheck: unnecessary now it's live
- Unskipped and fixed broken unit tests
   - Fixed session_info test for new result
   - Unskipped Tesscut tests now the service is live
- Converted all camelCase variables to under_score
   I probably missed some, but I tried to get them all.
- Added warning type to warnings in query.py: Not having an explicitly set warning type was making the warnings print oddly and cut off
- Added links to slides and STScI notebooks repository

bsipocz commented:
Wow @ceb8, you're the best, this looks to be a really nice cleanup. I'll try to give it a review soon.

bsipocz commented on astroquery/mast/core.py:
```diff
@@ -157,241 +155,101 @@ def __init__(self, username=None, password=None, session_token=None):
 
-        log.info("Authentication successful!\nSession Expiration: {}".format(exp))
-        return True
+    @deprecated(since="v0.3.10", message=("The get_token function is deprecated, "
```
I'm not sure what to do with this deprecation. Shall we do a release after this PR is merged? Otherwise it can go on for a long time before we get a tagged 0.3.10 so this deprecation would not necessarily mean much sense to users. @keflavich?

bsipocz replied:
@ceb8 - how long do you expect the deprecated way to be working?

ceb8 replied:
I am not sure I am doing this properly. Basically because of the way authentication works now, this function doesn't make much sense and is redundant and unnecessary since the "token" is now provided by the user, so there isn't a need for them to get it back. So I want to get rid of the function.  But it seems like it shouldn't just disappear, hence the deprecation warning. How would you recommend I handle this? I can do whatever you think makes the most sense.

bsipocz replied:
Hmm, I'm not sure either. If users were using this method, then a the warning or even an exception with an explaining message is indeed seems more useful than a generic AttributeError. However as it's not doing anything it probably wouldn't need to be a deprecation.

@keflavich - preferences/ideas?

ceb8 replied:
I changed the "since" argument to v0.3.9.dev, which will hopefully indicate to the user that is deprecated as of the version they are running.

bsipocz replied:
I'm changing it to 0.3.9 as it belongs to the rest of the deprecations made in that version. And then will look into to tag a 0.3.10 release.

bsipocz commented on astroquery/mast/core.py:
```diff
@@ -157,241 +155,101 @@ def __init__(self, username=None, password=None, session_token=None):
-
-        # clear any previous shib cookies
-        self._session.cookies.clear_session_cookies()
+        silent : bool, optional
```
We tend to have a `verbose` kwarg instead of `silent`, I think it would make sense to make it consistent.

ceb8 replied:
I agree. I'm trying to figure out the smoothest way to make this transition since `silent=False` == `verbose=True`. Does it make sense to have both arguments for a period of time where the use of "silent" triggers a deprecation warning?

bsipocz replied:
yes, and in fact there is a deprecation decorator in astropy to do this renaming: http://docs.astropy.org/en/latest/api/astropy.utils.decorators.deprecated_renamed_argument.html#astropy.utils.decorators.deprecated_renamed_argument

ceb8 replied:
Maybe I am missing an option, which would be great, but it looks to me like the two arguments need to **mean** the same thing to use that decorator. When in this case, to get the same behavior the user needs the opposite input.

bsipocz replied:
oh, yes, I suppose you're right. That case it can be deprecated for removal, and adding the verbose as a new one separately of the decorator. This very same decorator, with `new_name=None`.
And a short conditional to check for the case when both specified.

ceb8 replied:
Did it. The way decorating for removal appears to work is that the deprecated argument just doesn't get passed into the function anymore.  So that's why I didn't at the conditional check.

ceb8 replied:
I updated the decorator to put the info about the new argument to use in the 'since' argument. It's a bit hacky, but it means that it works with older versions of astropy utils now. And this won't be around forever.

ceb8 replied:
Still didn't work because "None" is not an acceptable new_name in the older version, so I pulled out the decorator and did the deprecation manually.  Not as tidy, but now it should work, and again the code won't be there forever.

bsipocz commented on astroquery/mast/core.py:
```diff
@@ -430,20 +288,20 @@ def _request(self, method, url, params=None, data=None, headers=None,
+        cur_page = 0
 
-        while curPage < totalPages:
+        while cur_page < total_pages:
```
wow, I haven't appreciated this before. Would be a nice feature request for upstream to ask to an API feature that returns everything without paging.

ceb8 replied:
Our new-style APIs don't require this, but the old style will persist.... for while. 😛

bsipocz replied:
oh, I wish all my wishes ever were working out this smoothly 😉

bsipocz commented on astroquery/mast/core.py:
```diff
@@ -1665,47 +1249,47 @@ def _download_files(self, products, base_dir, cache=True):
 
-            localPath = base_dir + "/" + dataProduct['obs_collection'] + "/" + dataProduct['obs_id']
-            dataUrl = self._MAST_DOWNLOAD_URL + "?uri=" + dataProduct["dataURI"]
+            local_path = base_dir + "/" + data_product['obs_collection'] + "/" + data_product['obs_id']
```
I wonder how this worked on windows before, please use `os.path.join()` instead.

bsipocz commented on astroquery/mast/core.py:
```diff
@@ -2121,54 +1702,54 @@ def download_hsc_spectra(self, spectra, download_dir=None, cache=True, curl_flag
 
-            localPath = download_dir.rstrip('/') + "/" + downloadFile + ".sh"
-            self._download_file(bundlerResponse['url'], localPath, head_safe=True)
+            local_path = download_dir.rstrip('/') + "/" + download_file + ".sh"
```
same, please use os.path.join

bsipocz commented:
@ceb8 - I have a few minor comments after which this seems to be ready to be merged.

ceb8 commented:
@bsipocz  While fixing up things in response to your comments, I noticed a few more things. So I've now removed all unused imports, and fixed the `requests.Response` mentions in the  docstrings so they properly link to the requests docs.

bsipocz commented:
Let me get back to it over the weekend to see how to fix the decorator for LTS.

ceb8 commented:
@bsipocz What do you think about my solutions to the two deprecation instances?

bsipocz commented:
Thank you @ceb8!
